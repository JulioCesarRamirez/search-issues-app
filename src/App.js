import React, { useState } from 'react';
import { GridItems } from './components/GridItems';
import { SearchItem } from './components/SearchItem';
import './styles.css';

export const App = ({ defaultItems = [] }) => {
  const [items, setItems] = useState(defaultItems);
  return (
    <div className="App">
      <div>
        <h3>Search Issues</h3>
        <SearchItem setItemToSearch={setItems} />
        <ul>
          {items.map((item, i) => (
            <GridItems key={i} {...item} />
          ))}
        </ul>
      </div>
    </div>
  );
};
