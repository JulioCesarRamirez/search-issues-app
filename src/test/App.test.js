
import React from "react";
import "@testing-library/jest-dom";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from "enzyme";
import { App } from "../App";

configure({adapter: new Adapter()});
describe("Test <App />", () => {
  it("Should show <App /> correctly", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should show a list of items  ", () => {
    const items = [{title:"react", url:'http...'}];
    const wrapper = shallow(<App defaultItems={items} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GridItems").length).toBe(items.length);
  });
});