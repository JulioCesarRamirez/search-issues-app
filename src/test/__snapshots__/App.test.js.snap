// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <App /> Should show <App /> correctly 1`] = `ShallowWrapper {}`;

exports[`Test <App /> should show a list of items   1`] = `ShallowWrapper {}`;
